relayer/src/channel.rs

```diff
@@ -74,14 +74,8 @@ mod handshake_retry {
     /// Translates from an error type that the `retry` mechanism threw into
     /// a crate specific error of [`ChannelError`] type.
     pub fn from_retry_error(e: retry::Error<ChannelError>, description: String) -> ChannelError {
-        match e {
-            retry::Error::Operation {
-                error: _,
-                total_delay,
-                tries,
-            } => ChannelError::max_retry(description, tries, total_delay),
-            retry::Error::Internal(reason) => ChannelError::retry_internal(reason),
-        }
+        let description = format!("{description}: {e}");
```

Comment:
@soareschen I wanted to use the underlying `ChannelError` included in the `retry::Error` as a source for `ChannelError::max_retry` but it seems that `flex-error` does not handle recursive error types, ie. one cannot specify `Box<ChannelError>` as an error source within an error variant of `ChannelError`. Do you see a way around that or is that a feature we could add to flex-error?

Reply:
You can use `[Self]` as an error source for recursive errors.

Reply:
Perfect, thanks! 🙏